crates/node/src/catchup.rs

```diff
@@ -373,11 +413,29 @@ impl Node {
                             last_transaction_hash: calimero_primitives::hash::Hash::default(),
                         };
 
-                        // We don't have the identity during catchup
-                        let initial_identity = KeyPair {
-                            public_key: PublicKey(*context_id),
-                            private_key: None,
-                        };
+                        // Retrieve the initial identity that should have been set by join_context
+                        let identity_key = calimero_store::key::ContextIdentity::new(
+                            context_id,
+                            PublicKey([0; 32]),
+                        );
+                        let initial_identity: KeyPair = self
+                            .store
+                            .handle()
+                            .get(&identity_key)?
+                            .ok_or_else(|| {
+                                eyre::eyre!("Initial identity not found for context {}", context_id)
+                            })?
+                            .into();
+
+                        // Share our public key with the catchup node
+                        self.network_client
+                            .publish(
+                                libp2p::gossipsub::TopicHash::from_raw(context_id),
+                                serde_json::to_vec(&types::PeerAction::SharePublicKey(
+                                    initial_identity.public_key.clone(),
+                                ))?,
+                            )
+                            .await?;
```

Comment:
nit here: this will announce this across the network and not just the catch up node